akka-cluster/src/main/scala/akka/cluster/ClusterDaemon.scala

```diff
@@ -657,10 +658,26 @@ private[cluster] class ClusterCoreDaemon(publisher: ActorRef) extends Actor with
           (remoteGossip, !remoteGossip.seenByNode(selfUniqueAddress), Newer)
         case _ ⇒
           // conflicting versions, merge
-          (remoteGossip merge localGossip, true, Merge)
+          val prunedLocalGossip = localGossip.members.foldLeft(localGossip) { (g, m) ⇒
```

Comment:
Ah, my fault, G2_pruned will not contain M anyway, since G2 did not contain it in the first place. This looks correct then.